third_party/Android.mk

```diff
@@ -85,13 +85,29 @@ include $(BUILD_STATIC_LIBRARY)
 
 SPVTOOLS_LOCAL_PATH := $(THIRD_PARTY_PATH)/spirv-tools
 LOCAL_PATH := $(SPVTOOLS_LOCAL_PATH)
+SPVTOOLS_OUT_PATH=$(abspath $(TARGET_OUT))
+
+define gen_spvtools_grammar_tables
+$(call generate-file-dir,$(SPVTOOLS_OUT_PATH)/opcode.inc)
+$(1)/opcode.inc $(1)/operand.inc: $(SPVTOOLS_LOCAL_PATH)/utils/generate_grammar_tables.py \
+                                  $(SPVTOOLS_LOCAL_PATH)/source/grammar.json
+		@$(HOST_PYTHON) $(SPVTOOLS_LOCAL_PATH)/utils/generate_grammar_tables.py \
+		                $(SPVTOOLS_LOCAL_PATH)/source/grammar.json \
+		                --opcode-file=$(1)/opcode.inc \
+		                --operand-file=$(1)/operand.inc
+		@echo "[$(TARGET_ARCH_ABI)] Grammar: opcode.inc & operand.inc <= grammar.json"
+$(SPVTOOLS_LOCAL_PATH)/source/opcode.cpp: $(1)/opcode.inc
+$(SPVTOOLS_LOCAL_PATH)/source/operand.cpp: $(1)/operand.inc
```

Comment:
I'm willing to let this go, but it does create a fairly strange build dependency, because now for the arm64 build depends on the x86 opcode.inc file.

The only thing I can thing that would solve this is as part of your build do something like copy $(SPVTOOLS_LOCAL_PATH)/source/opcode.cpp -> $(1)/opcode.cpp and change your build to use that opcode.cpp instead of $(SPVTOOLS_LOCAL_PATH)/source/opcode.cpp. 

Then arm64 would only depend on arm64 stuff. However that is very complicated, and I don't think will actually make any difference in practice.